lib/zhr_devs/submissions/commands/submit_solution.ex

```diff
@@ -0,0 +1,100 @@
+defmodule ZhrDevs.Submissions.Commands.SubmitSolution do
+  @moduledoc """
+  A command must contain a field to uniquely identify the aggregate instance (e.g. account_number).
+  Use @enforce_keys to force the identity field to be specified when creating the command struct.
+  """
+  @fields [:uuid, :hashed_identity, :technology, :task_uuid, :solution_path, :submission_identity]
+  @enforce_keys @fields
+  defstruct @fields
+
+  use Witchcraft.Comonad
+
+  alias Uptight.Result
+  alias Uptight.Text, as: T
+
+  alias ZhrDevs.App
+
+  alias ZhrDevs.Submissions.SubmissionIdentity
+
+  import Uptight.Assertions
+
+  @supported_technologies Enum.map(
+                            Application.compile_env(:zhr_devs, :supported_technologies),
+                            &Atom.to_string/1
+                          )
+
+  @typep error() :: String.t() | struct()
+
+  @spec dispatch(Keyword.t()) :: :ok | {:error, error()}
+  def dispatch(opts) do
+    case parse(opts) do
+      %Uptight.Result.Ok{} = ok_result ->
+        ok_result
+        |> Result.from_ok()
+        |> App.dispatch()
+
+      error ->
+        {:error, error |> extract() |> build_error()}
```

Comment:
Does some library want us to have errors shaped exactly like this? Remember, trace part of Result is very important for enabling debugging.

Reply:
So it's better to return error here as it is and then perform this unpacking of exact error close to the presentation layer?

Reply:
Of course. We should preserve as much data as possible and let the consumers figure out how to display.